src/main/java/UserInterface.java

```diff
@@ -0,0 +1,223 @@
+        } catch (Exception e) {
+            System.out.println("Error: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Constructs a subclass of Task from an ArrayList<String>.
+     * @param taskStringArray the ArrayList<String> containing the user's input, split up by whitespace.
+     *                        The first element of this list should contain the subclass of task to be constructed.
+     * @return a subclass of Task determined by the first element of taskStringArray.
+     */
+    private Task buildTask(ArrayList<String> taskStringArray) {
+        String taskType = taskStringArray.get(0);
+        ArrayList<StringBuilder> taskParts = splitTask(taskStringArray);
+        String prepositions = String.valueOf(taskParts.get(taskParts.size() - 1));
+        switch (taskType) {
+        case COMMAND_TODO:
+            return new ToDo(String.valueOf(taskParts.get(0)));
+        case COMMAND_DEADLINE:
+            return new Deadline(String.valueOf(taskParts.get(0)), String.valueOf(taskParts.get(1)), prepositions);
+        case COMMAND_EVENT:
+            return new Event(String.valueOf(taskParts.get(0)), String.valueOf(taskParts.get(1)), prepositions);
+        }
+        return null;
+    }
+
+    /**
+     * Splits an ArrayList<String> into an ArrayList<StringBuilder>, where each element
+     * can be used to construct a subclass of Task. Use this in conjunction with buildTask to construct a
+     * subclass of Task.
+     * @param input ArrayList of Strings that represent the user's input split by whitespace.
+     * @return an ArrayList<StringBuilder> containing the input split by the '/' character.
+     *         Each element represents a String used to construct the task.
+     *         The last element contains all prepositions used in a single string.
+     *         e.g.: deadline return book /by Sunday /at library returns the following Arraylist:
+     *         {"return book", "Sunday", "Library", "/by/at"}
+     */
+    private ArrayList<StringBuilder> splitTask(ArrayList<String> input) {
+        /* taskPartsIterator increments when a String containing '/' is encountered.
+         * inputIterator increments after each String in input is read.
+         * The loop ends when inputIterator exceeds the size of the ArrayList, indicating that all
+         * Strings in input have been read.
+         */
+        ArrayList<StringBuilder> taskParts = new ArrayList<>();
+        StringBuilder prepositions = new StringBuilder();
+        for (int taskPartsIterator = 0, inputIterator = 0; inputIterator < input.size(); ++taskPartsIterator) {
+            String finalWord = input.get(inputIterator);
+            if (finalWord.startsWith("/")) {
+                prepositions.append(finalWord);
+            }
+            inputIterator++;
+            taskParts.add(new StringBuilder());
+            for (; inputIterator < input.size() && !input.get(inputIterator).startsWith("/"); ++inputIterator) {
```

Comment:
Can consider splitting lines after the '&&'

Reply:
Thank you. I will implement this.